File: aula04-hw02-optional-args/Point.java
import java.lang.Math;

public class Point{
    private final int x, y;
    
    private static final Point root = new Point(0,0);

    Point(int a, int b){
        this.x = a;
        this.y = b;
    }

    public int getX() {return this.x;}

    public int getY() {return this.y;}

    @Override
    public String toString() {
        return "Point(x=" + x + ", y=" + y +")";
    }

    public double distance(Point other ){
        int dX = this.x - other.x;
        int dY = this.y - other.y;
        return Math.sqrt(dX*dX + dY*dY);
    }

    public double distance(){
        return distance(root);
    }
}

File: aula06-logger/src/test/kotlin/pt/isel/Point.kt
package pt.isel

import kotlin.math.sqrt

data class Point(val x: Int, val y: Int) {
    fun distance(other: Point = Point(0,0)) : Double {
        val dX = x - other.x
        val dY = y - other.y
        return sqrt((dX*dX + dY*dY).toDouble())
    }
}
